### ISSUE TITLE
How you managed to write complementary GAN please

### ISSUE THREAD

**Issue by Avv22** [salty-vanilla/tf-anomaly#2]
Fantastic Work,

How long did it take from you to write the code please? Is the complementary GAN same as the one in the paper please? Why you decided to do it in tf and not keras please?

**Comment by salty-vanilla** [comment]
Hi @Avra2 
Sorry for my late reply.

> How long did it take from you to write the code please?

It takes a day to write the code. (per paper)

> Is the complementary GAN same as the one in the paper please?

My code of Complementary GAN is not same  as the paper.
I use Conv Layers to apply to image anomaly detection.

> Why you decided to do it in tf and not keras please?

I was used to tf, and it was time to update to tf2 then.
It's hard to implement complex processing with keras alone, so I must use backend(tf) .

**Comment by Avv22** [comment]
@salty-vanilla . Thank you very much for replying back to me. I guess this is where you did your the complementary idea in GAN please? 

`loss_kl = tf.reduce_mean(tf.boolean_mask(tf.math.log(tf.nn.sigmoid(d_fake)+self.eps), tf.nn.sigmoid(d_fake) > self.threshold))`
